improve integration doc (#9322)
hibernate/hibernate-orm: documentation/src/main/asciidoc/integrationguide/chapters/services/Services.adoc

The basic requirement for a `Service` is to implement the marker interface `org.hibernate.service.Service`.
Hibernate uses this internally for some basic type safety.

> The `Service` can also implement a number of optional lifecycle related contracts:

Author: change this to maintain consistency. `lifecycle` is used elsewhere

The `Service` can also implement a number of optional lifecycle related contracts:

`org.hibernate.service.spi.Startable`::
 	allows the `Service` impl to be notified that it is being started and about to be put into use.
`org.hibernate.service.spi.Stoppable`::
	allows the `Service` impl to be notified that it is being stopped and will be removed from use.
`org.hibernate.service.spi.ServiceRegistryAwareService`::
	allows the `Service` to be injected with a reference to the registry that is managing it. See <<services-dependencies>> for more details.
`org.hibernate.service.spi.Manageable`::

> 	marks the `Service` as manageable in JMX provided the JMX integration is enabled. This feature is still incomplete.

Author: The above stuff has been absent from the codebase

Any method on the `Service` implementation class accepting a single parameter and annotated with `@InjectService` is considered requesting injection of another service.
+
By default, the type of the method parameter is expected to be the `Service` role to be injected.

> If the parameter type is different than the `Service` role, the `serviceRole` attribute of the `@InjectService` annotation should be used to explicitly name the role.

Reviewer: ```suggestion
If the parameter type is different to the `Service` role, the `serviceRole` attribute of the `@InjectService` annotation should be used to explicitly name the role.
```

+

> By default, injected services are considered required, that is the startup will fail if a named dependent `Service` is missing.

Reviewer: ```suggestion
By default, injected services are considered required, that is, the startup will fail if a named dependent `Service` is missing.
```

=== Types of ServiceRegistries

Currently Hibernate utilizes three different `ServiceRegistry` implementations forming a hierarchy.

> Each type is a specialization for the purpose of type safety, but they add no new functionality.

Author: for consistency reason. `type safety` is used more often elsewhere.

[[services-registry-boot]]
==== `BootstrapServiceRegistry`

> The `org.hibernate.boot.registry.BootstrapServiceRegistry` holds three `Services` and is normally built by means of the `org.hibernate.boot.registry.BootstrapServiceRegistryBuilder` factory class.

Reviewer: ```suggestion
The `org.hibernate.boot.registry.BootstrapServiceRegistry` holds three ``Service``s and is normally built by means of the `org.hibernate.boot.registry.BootstrapServiceRegistryBuilder` factory class.
```

The specific capabilities exposed on this `Service` include:

* Locating `Class` references by name. This includes application classes as well as _integration_ classes.

> * Locating resources (properties files, xml files, etc.) as _classpath resources_.

Reviewer: ```suggestion
* Locating resources (properties files, xml files, and so on) as _classpath resources_.
```

* Interacting with `java.util.ServiceLoader`, Java's own `Service` provider discovery mechanism.

===== `IntegratorService`

The `Service` role for this `Service` is `org.hibernate.integrator.spi.IntegratorService.` 

> Applications, third-party integrators and others all need to integrate with Hibernate. Historically this used to require something (usually the application) to coordinate registering the pieces of each integration needed on behalf of each integration. The `org.hibernate.integrator.spi.Integrator` contract formalized this "integration SPI". The `IntegratorService` manages all known integrators.

Reviewer: ```suggestion
Applications, third-party integrators, and others all need to integrate with Hibernate. Historically this used to require something (usually the application) to coordinate registering the pieces of each integration needed on behalf of each integration. The `org.hibernate.integrator.spi.Integrator` contract formalized this "integration SPI". The `IntegratorService` manages all known integrators.
```

Oxford comma.
